Made some much needed changes to the ReadMe (#8372)
TeamNewPipe/NewPipe: README.md

<p align="center"><a href="https://newpipe.net"><img src="assets/new_pipe_icon_5.png" width="150"></a></p> 
<h2 align="center"><b>NewPipe</b></h2>

> <h4 align="center">A libre lightweight streaming front-end for Android.</h4>

Reviewer: Both the original and your suggestion are correct, though I prefer the one with a hyphen.

</p>
<hr>
<p align="center"><a href="#screenshots">Screenshots</a> &bull; <a href="#supported-services">Supported Services</a> &bull; <a href="#description">Description</a> &bull; <a href="#features">Features</a> &bull; <a href="#installation-and-updates">Installation and updates</a> &bull; <a href="#contribution">Contribution</a> &bull; <a href="#donate">Donate</a> &bull; <a href="#license">License</a></p>
<p align="center"><a href="https://newpipe.net">Website</a> &bull; <a href="https://newpipe.net/blog/">Blog</a> &bull; <a href="https://newpipe.net/FAQ/">FAQ</a> &bull; <a href="https://newpipe.net/press/">Press</a></p>
<hr>

*Read this in other languages: [English](README.md), [Español](doc/README.es.md), [हिन्दी](doc/README.hi.md), [한국어](doc/README.ko.md), [Soomaali](doc/README.so.md), [Português Brasil](doc/README.pt_BR.md), [Polski](doc/README.pl.md), [日本語](doc/README.ja.md), [Română](doc/README.ro.md), [Türkçe](doc/README.tr.md), [正體中文](doc/README.zh_TW.md).*

<b>WARNING: THIS APP IS IN BETA, SO YOU MAY ENCOUNTER BUGS. IF YOU DO, OPEN AN ISSUE IN OUR GITHUB REPOSITORY BY FILLING OUT THE ISSUE TEMPLATE.</b>

> <b>PUTTING NEWPIPE, OR ANY FORK OF IT, INTO THE GOOGLE PLAY STORE VIOLATES THEIR TERMS AND CONDITIONS.</b>

Reviewer: What's the reasoning behind this change? It was already correct, and I prefer the original version too.

Author: Commas to introduce pause, and highlight that even forks aren't allowed.

[<img src="fastlane/metadata/android/en-US/images/tenInchScreenshots/shot_11.png" width=405>](fastlane/metadata/android/en-US/images/tenInchScreenshots/shot_11.png)
[<img src="fastlane/metadata/android/en-US/images/tenInchScreenshots/shot_12.png" width=405>](fastlane/metadata/android/en-US/images/tenInchScreenshots/shot_12.png)

### Supported Services

NewPipe currently supports these services:

> * YouTube ([website](https://www.youtube.com/)) and YouTube Music ([website](https://music.youtube.com/)) ([wiki](https://en.wikipedia.org/wiki/YouTube))
> * PeerTube ([website](https://joinpeertube.org/)) and all its instances (open the website to know what that means!) ([wiki](https://en.wikipedia.org/wiki/PeerTube))
> * Bandcamp ([website](https://bandcamp.com/)) ([wiki](https://en.wikipedia.org/wiki/Bandcamp))
> * SoundCloud ([website](https://soundcloud.com/)) ([wiki](https://en.wikipedia.org/wiki/SoundCloud))
> * media.ccc.de ([website](https://media.ccc.de/)) ([wiki](https://en.wikipedia.org/wiki/Chaos_Computer_Club))

Reviewer: ```suggestion
* [YouTube](https://www.youtube.com/) and [YouTube Music](https://music.youtube.com/) ([Wikipedia](https://en.wikipedia.org/wiki/YouTube))
* [PeerTube](https://joinpeertube.org/) and all its instances (open the website to know what that means!) ([Wikipedia](https://en.wikipedia.org/wiki/PeerTube))
* [Bandcamp](https://bandcamp.com/) ([Wikipedia](https://en.wikipedia.org/wiki/Bandcamp))
* [SoundCloud](https://soundcloud.com/) ([Wikipedia](https://en.wikipedia.org/wiki/SoundCloud))
* [media.ccc.de](https://media.ccc.de/) ([Wikipedia](https://en.wikipedia.org/wiki/Chaos_Computer_Club))
```

Author: That's what I initially did, but I didn't want anyone even slightly unawares to accidentally open a website they didn't want to, e.g. YT. So I made the links explicitly clear.

Reviewer: ```suggestion
<!-- We link to the service websites separately to avoid people accidentally opening a website they didn't want to. -->
* YouTube ([website](https://www.youtube.com/)) and YouTube Music ([website](https://music.youtube.com/)) ([wiki](https://en.wikipedia.org/wiki/YouTube))
* PeerTube ([website](https://joinpeertube.org/)) and all its instances (open the website to know what that means!) ([wiki](https://en.wikipedia.org/wiki/PeerTube))
* Bandcamp ([website](https://bandcamp.com/)) ([wiki](https://en.wikipedia.org/wiki/Bandcamp))
* SoundCloud ([website](https://soundcloud.com/)) ([wiki](https://en.wikipedia.org/wiki/SoundCloud))
* media.ccc.de ([website](https://media.ccc.de/)) ([wiki](https://en.wikipedia.org/wiki/Chaos_Computer_Club))
```

As you can see, NewPipe supports multiple video and audio services. Though it started off with YouTube, other people have added more services over the years, making NewPipe more and more versatile!

> Partially due to circumstance, and partially due to its popularity, YouTube is the best supported out of these services. If you use or are familiar with any of these other services, please help us improve support for them! We're looking for maintainers for Soundcloud and PeerTube.

Reviewer: ```suggestion
Partially due to circumstance, and partially due to its popularity, YouTube is the best supported out of these services. If you use or are familiar with any of these other services, please help us improve support for them! We're looking for maintainers for SoundCloud and PeerTube.
```

If you intend to add a new service, please get in touch with us first! Our [docs](https://teamnewpipe.github.io/documentation/) provide more information on how a new service can be added to the app and to the [NewPipe Extractor](https://github.com/TeamNewPipe/NewPipeExtractor).

## Description

NewPipe works by fetching the required data from the official API (e.g. PeerTube) of the service you're using. If the official API is restricted (e.g. YouTube) for our purposes, or is proprietary, the app parses the website or uses an internal API instead. This means that you don't need an account on any service to use NewPipe.

Also, since they are free and open source software, neither the app nor the Extractor use any proprietary libraries or frameworks, such as Google Play Services. This means you can use NewPipe on devices or custom ROMs that do not have Google apps installed.

### Features

> * Watch videos and listen to audios (yes, you can listen to YouTube in the background; it will only load and play the audio stream! Data-saving FTW!) 

Reviewer 2: Think thats too much for a readme

Author: We don't have in-app tours or anything of the sort. This ReadMe is the only place for a user to discover features other than messing around in the app themselves. Most users don't have that sort of mindset.

Reviewer: I agree with @Poolitzer, this is a little too long. Maybe this is better?

```suggestion
* Watch videos at resolutions up to 4K
* Listen to audio in the background, only loading the audio stream to save data
```

Author: These were originally separate points. We could just go that route, with some modification.

Reviewer: @opusforlife2 I think my above suggestion is good, what do you think?

> * Popup mode (floating player, aka Picture-in-Picture) (for all you multi-taskers out there)

Reviewer 2: same

Author: Reply above ^

Reviewer: I'm okay with jokes, but this one feels maybe a little "forced" and a bit too long? We're trying to be brief here.

Author: What joke?

> We're trying to be brief here.

Why?

Reviewer: "for all you multi-taskers out there"

> Why?

I feel that the features list should just be short and to the point, with no fluff.

Reviewer: @opusforlife2 so, remove it?

```suggestion
* Popup mode (floating player, aka Picture-in-Picture)
```

* Watch live streams
* Show/hide subtitles/closed captions
* Search videos and audios (on YouTube, you can specify the content language as well)
* Enqueue videos (and optionally save them as local playlists)
* Show/hide general information about videos (such as description and tags)
* Show/hide next/related videos
* Show/hide comments

> * 1080p/2K/4K support

Reviewer 2: hm, that makes it seem like we dont support 720p etc. Probably would delete that.

Author: Support for multiple resolutions, up to 4K

Reviewer: I would put this with the first bullet point, see my suggestion above.

Author: Replied ^

Author: Whoops. This escaped the previous commit.

```suggestion
```

* Search videos, audios, channels, playlists and albums
* Browse videos and audios within a channel

> * Subscribe to channels (yes, without logging into any account!)

Reviewer 2: me still no likey jokes here.

Author: Wut

Reviewer: @Poolitzer why not?

Author: Again, what joke?

Reviewer: @Poolitzer probably meant the phrase in parentheses. I'd say to keep it, as it points out to the reader that login is not needed to subscribe to channels.

* Get notifications about new videos from channels you're subscribed to
* Create and edit channel groups (for easier browsing and management)
* Browse video feeds generated from your channel groups
* View and search your watch history
* Search and watch playlists (these are remote playlists, which means they're fetched from the service you're browsing)
* Create and edit local playlists (these are created and saved within the app, and have nothing to do with any service)

> * Download videos
> * Download audios
> * Download subtitles/closed captions

Reviewer 2: Put em in one point.
```suggestion
* Download videos/audios/subtitles (closed captions)
```

* Open in Kodi
* Watch/Block age-restricted material

> * Orbot/Tor support (not within the app yet, but works when used system-wide)

Reviewer 2: But I mean, doesn't every app?

Author: Some apps leak info despite having Orbot enabled, I think.

Author: But it would be good if someone with the knowledge chimes in.

Reviewer: I think there is no reason to mention this, as we provide no special integration with Orbot or Tor.

https://www.stoutner.com/webrtc/ may be relevant.

Author: Alright. Let's remove it.

Reviewer: @opusforlife2
```suggestion
```

You can install NewPipe using one of the following methods:
 1. Add our custom repo to F-Droid and install it from there. The instructions are here: https://newpipe.net/FAQ/tutorials/install-add-fdroid-repo/
 2. Download the APK from [GitHub Releases](https://github.com/TeamNewPipe/NewPipe/releases) and install it.
 3. Update via F-Droid. This is the slowest method of getting updates, as F-Droid must recognize changes, build the APK itself, sign it, and then push the update to users.
 4. Build a debug APK yourself. This is the fastest way to get new features on your device, but is much more complicated, so we recommend using one of the other methods.
 5. If you're interested in a specific feature or bugfix provided in a Pull Request in this repo, you can also download its APK from within the PR. Read the PR description for instructions. The great thing about PR-specific APKs is that they're installed side-by-side the official app, so you don't have to worry about losing your data or messing anything up.

> We recommend method 1 for most users. APKs installed using method 1 or 2 are compatible with each other (meaning you can update your app using either method), but not with those installed using method 3. This is due to the same signing key (ours) being used for 1 and 2, but a different signing key (F-Droid's) being used for 3. Building a debug APK using method 4 excludes a key entirely. Signing keys help ensure that a user isn't tricked into installing a malicious update to an app. When using method 5, each APK is signed with a different random key supplied by GitHub Actions, so you cannot even update it. You will have to backup and restore the app data each time you wish to use a new APK.

Reviewer 2: Dont think the last sentence is relevant for the average reader.

Author: We don't mention this anywhere else.

Reviewer: > We recommend method 1 for most users. APKs installed using method 1 or 2 are compatible with each other (meaning you can update your app using either method)

The phrase in the parenthesis is somewhat confusing, as it gives the idea that you can update an APK installed by _any_ method using method 1 or 2, which is not true.

Author: Good point, that part should be reworded.

Reviewer: Other suggestions or improvements to mine appreciated, this is the best I could come up with.
```suggestion
We recommend method 1 for most users. APKs installed using method 1 or 2 are compatible with each other (meaning that if you installed NewPipe using either method 1 or 2, you can also update NewPipe using the other), but not with those installed using method 3. This is due to the same signing key (ours) being used for 1 and 2, but a different signing key (F-Droid's) being used for 3. Building a debug APK using method 4 excludes a key entirely. Signing keys help ensure that a user isn't tricked into installing a malicious update to an app. When using method 5, each APK is signed with a different random key supplied by GitHub Actions, so you cannot even update it. You will have to backup and restore the app data each time you wish to use a new APK.
```

In the meanwhile, if you want to switch sources for some reason (e.g. NewPipe's core functionality breaks and F-Droid doesn't have the latest update yet), we recommend following this procedure:
1. Back up your data via Settings > Content > Export Database so you keep your history, subscriptions, and playlists
2. Uninstall NewPipe
3. Download the APK from the new source and install it
4. Import the data from step 1 via Settings > Content > Import Database

<b>Note: when you're importing a database into the official app, always make sure that it is the one you exported _from_ the official app. If you import a database exported from an APK other than the official app, it may break things. Such an action is unsupported, and you should only do so when you're absolutely certain you know what you're doing.</b>

## Contribution

> Whether you have ideas, translations, design changes, code cleaning, or even major code changes, help is always welcome. The more is done the better it gets! If you'd like to get involved, check our [contribution notes](.github/CONTRIBUTING.md).

Reviewer 2: ```suggestion
Whether you have ideas, translations, design changes, code cleaning, or even major code changes, help is always welcome. The more getting done the better it gets! If you'd like to get involved, check our [contribution notes](.github/CONTRIBUTING.md).
```

Author: Well, that is also incorrect. Let's change it completely.

"The app gets better and better with each contribution, no matter how big or small!"

Reviewer: What's wrong grammatically with "the more is done the betteer it gets"?

Author: Nothing, but it does read a bit weird. Let's replace it anyway.

Reviewer: ```suggestion
Whether you have ideas, translations, design changes, code cleaning, or even major code changes, help is always welcome. The app gets better and better with each contribution, no matter how big or small! If you'd like to get involved, check our [contribution notes](.github/CONTRIBUTING.md).
```

<a href="https://hosted.weblate.org/engage/newpipe/">
<img src="https://hosted.weblate.org/widgets/newpipe/-/287x66-grey.png" alt="Translation status" />
</a>

## Donate

> If you like NewPipe, you're welcome to send a donation. We prefer Liberapay, as it is both open source and non-profit. For further info on donating to NewPipe, please visit our [website](https://newpipe.net/donate).

Reviewer: ```suggestion
If you like NewPipe, you're welcome to send a donation. We prefer Liberapay, as it is both open-source and non-profit. For further info on donating to NewPipe, please visit our [website](https://newpipe.net/donate).
```
"open-source" is the adjective. See https://en.wiktionary.org/wiki/open-source#English and https://en.wiktionary.org/wiki/open_source#English.
